core/src/test/java/org/elasticsearch/recovery/RecoveryWhileUnderLoadIT.java

```diff
@@ -255,19 +256,21 @@ public void testRecoverWhileRelocating() throws Exception {
             logger.info("--> refreshing the index");
             refreshAndAssert();
             logger.info("--> verifying indexed content");
-            iterateAssertCount(numShards, indexer.totalIndexedDocs(), 10);
+            iterateAssertCount(numShards, (int) indexer.totalIndexedDocs(), 10);
         }
     }
 
     private void iterateAssertCount(final int numberOfShards, final long numberOfDocs, final int iterations) throws Exception {
         SearchResponse[] iterationResults = new SearchResponse[iterations];
         boolean error = false;
+        SearchResponse lastErroneousResponse = null;
         for (int i = 0; i < iterations; i++) {
-            SearchResponse searchResponse = client().prepareSearch().setSize(0).setQuery(matchAllQuery()).get();
+            SearchResponse searchResponse = client().prepareSearch().setSize((int) numberOfDocs).setQuery(matchAllQuery()).addSort("id", SortOrder.ASC).get();
```

Comment:
I'm missing the point of this casts; the corresponding method parameter is a `long`?